File: 0643-Maximum Average Subarray I/0643-Maximum Average Subarray I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        int i = 0;
        int sum = 0;
        while (i < k)
        {
            sum += nums[i++];
        }
        
        int maxSum = sum;
        while (i < n)
        {
            sum += nums[i] - nums[i - k];
            maxSum = max(maxSum, sum);
            ++i;
        }
        
        return (double)maxSum / k;
    }
};

File: 0601-0700/0643-Maximum Average Subarray I/0643-Maximum Average Subarray I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        s = sum(nums[:k])
        maxSum = s
        for i in range(k, len(nums)):
            s += nums[i] - nums[i - k]
            maxSum = max(maxSum, s)
        return maxSum / k
